--- original/src/libespeak-ng/soundicon.c	2022-11-01 10:06:27.000000000 +0100
+++ src/libespeak-ng/soundicon.c	2022-11-04 19:25:08.000000000 +0100
@@ -182,7 +182,8 @@
 		return -1;
 
 	soundicon_tab[n_soundicon_tab].filename = (char *)realloc(soundicon_tab[n_soundicon_tab].filename, strlen(fname)+1);
-	strcpy(soundicon_tab[n_soundicon_tab].filename, fname);
+	if (soundicon_tab[n_soundicon_tab].filename)
+		strcpy(soundicon_tab[n_soundicon_tab].filename, fname);
 	n_soundicon_tab++;
 	return n_soundicon_tab - 1;
 }
