--- original/src/libespeak-ng/spect.c	2022-11-04 19:17:03.000000000 +0100
+++ src/libespeak-ng/spect.c	2022-11-04 19:25:08.000000000 +0100
@@ -25,7 +25,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+// for BYTE_ORDER, BIG_ENDIAN
+#if HAVE_ENDIAN_H
 #include <endian.h>
+#endif
+
 
 #include <espeak-ng/espeak_ng.h>
 #include <espeak-ng/speak_lib.h>
@@ -348,6 +352,10 @@
 		free(spect->frames);
 	}
 	spect->frames = calloc(n, sizeof(SpectFrame *));
+	if (spect->frames==NULL){
+		fprintf(stderr, "SpectFrame: Out of memory.\n");
+		return ENOMEM;
+	}
 
 	spect->numframes = 0;
 	spect->max_x = 3000;
