--- original/src/libespeak-ng/voices.c	2022-11-01 10:06:27.000000000 +0100
+++ src/libespeak-ng/voices.c	2022-11-04 19:28:08.000000000 +0100
@@ -32,7 +32,7 @@
 #if defined(_WIN32) || defined(_WIN64)
 #include <windows.h>
 #else
-#include <dirent.h>
+#include "direntx.h"
 #endif
 
 #include <espeak-ng/espeak_ng.h>
@@ -223,6 +223,8 @@
 		return NULL; // no language lines in the voice file
 
 	p = (char *)calloc(sizeof(espeak_VOICE) + langix + strlen(fname) + strlen(vname) + 3, 1);
+	if (p==NULL) 
+		return NULL;
 	voice_data = (espeak_VOICE *)p;
 	p = &p[sizeof(espeak_VOICE)];
 
